app/controllers/user.js

```diff
@@ -16,12 +16,38 @@ const passwordValid = password => {
   return isAlphanumeric.test(password) && password.length >= 8;
 };
 
+const generateUser = (user, next) => {
+  if (!emailValid(user.email)) {
+    logger.error(`Email: ${user.email} invalid.`);
+    return next(errors.emailNotValid(user.email));
+  }
+  if (!passwordValid(user.password)) {
+    logger.error('Password invalid.');
+    return next(errors.passwordInvalid);
+  }
+  logger.info(`All validations passed, going to create the user: ${JSON.stringify(user)}`);
+  return bcrypt.hash(user.password, saltRounds).then(hash => {
+    user.password = hash;
+    return User.createModel(user);
+  });
+};
+const giveAdminPriviledges = user => {
+  return User.update({ admin: true }, { returning: true, where: { email: user.email } });
```

Comment:
Here you are receiving the userParams and not the user thats why you need to make this query, but if you receive the actual user instance you can change it to `return user.update({ admin: true });` also you might have some concurrency problems with the current solution.